sessions/04-using-delay-distributions-to-model-the-data-generating-process-of-an-epidemic/index.qmd

```diff
@@ -2,3 +2,291 @@
 ---
 
+```{r echo = FALSE}
+set.seed(123)
+```
+
+# Objectives
+
+The aim of this session is to introduce how delay distributions can be used to model the population-level data generating process of an epidemic.
+
+# Libraries used
+
+In this session we will use the `nfidd` package to load the data set of infection times, the `dplyr` and `tidyr` packages for data wrangling, the `lubridate` package to handle dates, `ggplot2` library for plotting, the `here` library to find the stan model, and the `cmdstanr` library for using stan.
+We will also use the `tidybayes` package for extracting results of the inference.
+
+```{r libraries, message = FALSE}
+library("nfidd")
+library("dplyr")
+library("tidyr")
+library("lubridate")
+library("ggplot2")
+library("here")
+library("cmdstanr")
+library("tidybayes")
+```
+
+::: {.callout-tip}
+The code in this session can be run as an interactive notebook using RStudio, or copied-and-pasted into an R session.
+It needs to be run inside the course repository so that the `here()` commands below find the stan model files.
+:::
+
+# Simulating observations from a time series of infections
+
+As above we first simulate the process that generates the data, before using the same model to conduct inference.
+
+## Delay distributions and convolutions
+
+In the last session we simulated _individual outcomes_ from a delay distribution, and then re-estimated the corresponding parameters.
+However, sometimes we do not have data on these individual-level outcomes, either because they are not recorded or because they cannot be shared, for example due to privacy concerns.
+At the population level, individual level delays translate into _convolutions_.
+If we have a time series of infections $I_t$ ($t=1, 2, 3, \ldots, t_\mathrm{max}$), where $t$ denotes the day on which the infections occur, and observable outcomes occur with a delay given by a delay distribution $p_i$ ($i=0, 1, 2, \dots, p_\mathrm{max}$), where $i$ is the number of days after infection that the observation happens, then the number of observable outcomes $C_t$ on day $t$ is given by
+
+$$
+C_t = \sum_{i=0}^{i=p_\mathrm{max}} I_{t-i} p_i
+$$
+
+In words, to get the number of observable outcomes on day $t$ is given by the sum of infections on all previous days multiplied by the probability that those infections are observed on day $t$.
+For example, the observable outcomes $C_t$ could be the number of symptom onsets on day $t$ and $p_i$ is the incubation period.
+
+We can use exactly the same data as in the previous session but this time aggregate it into a time series:
+
+```{r aggregate}
+data(infection_times)
+## assume a start date, e.g. 10 May of this year
+start_date <- ymd("2024-05-10")
+df <- infection_times |>
+  mutate(
+    incubation_period = rgamma(n(), shape = 5, rate = 1),
+    infection_date = floor_date(start_date + infection_time, unit = "day"),
+    onset_date = floor_date(
```

Comment:
I think in an earlier session we could introduce a function to do this times -> date conversion and then use it? That way people can understand the operation as part of the censoring session and not have it as visual clutter as we go on?

Reply:
I have moved away from dates now, to be introduced in #54 instead.